tour_model_eval/Evaluate sim wrt filtration and different radii - unrolled.ipynb

```diff
@@ -0,0 +1,1191 @@
+    "\n",
+    "The main reasons why these are identical is because the distributions appear to be long-tailed, so the cutoff step basically only deletes 1 trip clusters. If this were not true, there would actually be a difference in the noisy trips between the two approaches and a difference in the results.\n",
+    "\n",
+    "However, the choice of cutting off **can potentially** affect the modeling. If 50% of the trips are below the cutoff (as we saw in the similarity deep dive, we have a much smaller pool of entries to work with. And if one of the test entries does in fact match the 50% of the trips that are deleted, we won't have to ask the user.\n",
+    "\n",
+    "To capture this, we need to build a model with or without the cutoff and match **something else** to model. That will give us an idea of whether those infrequent trips really do matter when matching with trips not in the model.\n",
+    "\n",
+    "I can think of two options for **something else**.\n",
+    "- Split the data: we should check the performance with both a large model and small data (state for each iteration) and a small model and large data (state overall/when we reset the pipeline).\n",
+    "- Use the unlabeled data: while the unlabeled data doesn't have labels, it certainly has start and end points. We should be able to \n",
+    "\n",
+    "\n",
+    "So it seems like the homogeneity score is a property of the model (which makes sense since it requires labeled data), and the request pct can/should be a property of the **unlabeled data** tested against that model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4e88bc20",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "h_score_no_na(expanded_trip_df_map[all_users[0]].no_filter_no_cutoff_100, expanded_trip_df_map[all_users[0]].no_filter_cutoff_100)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94a0e74b",
+   "metadata": {},
+   "source": [
+    "## Experiment with a new version of the function that treats trips below cutoff as single trip clusters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1491241f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reimplemented version\n",
+    "def h_score_no_na(labels_true, labels_pred):\n",
+    "    na_index = labels_true[pd.isna(labels_true)].index\n",
+    "    pred_noise_index = labels_pred[labels_pred < 0].index\n",
+    "    # Copy before we set the index to nan; we don't want to have a side effect here!\n",
+    "    new_labels_true = labels_true.copy()\n",
+    "    new_labels_pred = labels_pred.copy()\n",
+    "    # Create a set of unique clusters for the noisy trips\n",
+    "    new_labels_pred.loc[pred_noise_index] = range(len(pred_noise_index))\n",
```

Comment:
I think the `new_labels_pred` for noise would overlap the one for trips that are not noise.
I would use the number in len(trip_df) that are not used by clustered trips.
So, I think the result from here is not correct.

Reply:
Can you actually generate a use case for which this is wrong?

Reply:
For example, range(len(pred_noise_index)) begins from 0, then first noise predicted label would be 0. In clustered trips, the first cluster also has label 0.
I didn't get error message. I just think the way to assign the label may be not correct. Just want to know if I make any mistake.

Reply:
that's a good point!

I wasn't asking about an error message; I was thinking more along the lines of a small code snippet/unit test showing what is wrong. I think you are right, although it would be good to confirm, and with a small unit test, we could also verify that the fix is correct.

I am not sure I understand what you mean by "I would use the number in len(trip_df) that are not used by clustered trips." or how that translates to code.